Plugins~/Src/MeshSyncClientBlender/msblenContext.cpp

```diff
@@ -1448,10 +1449,19 @@ bool msblenContext::sendObjects(MeshSyncClient::ObjectScope scope, bool dirty_al
     }
 
 #if BLENDER_VERSION >= 300
-    if (m_geometryNodeUtils.getInstancesDirty() || dirty_all) {
+    // The dependency graph update event does not fire when the scene changes while the timeline is playing.
+    // To ensure geometry nodes get exported correctly while playing, check the frame number:
+    static int lastExportedFrame = scene.GetCurrentFrame();
```

Comment:
Confirming.
Is this assignment of `scene.GetCurrentFrame` an optimization to skip export when doing the first sync ?
I am wondering why we are not simply assigning -1 or something else into `lastExportedFrame`.

Reply:
It could probably be anything because the first sync will be a full sync anyway. I used GetCurrentFrame() to initialize it to the same value as the currentFrame variable.

Reply:
If that is the case, can't we just set it with -1 to: 
1. clearly indicate that the initial sync is supposed to be a full sync 
2. avoid unnecessary function call

Reply:
Ok done